lib/core/src/rcMisc.cpp

```diff
@@ -1566,17 +1566,37 @@ getLocalTimeStr( struct tm *mytm, char *timeStr ) {
 }
 
 
-/* Get the current time + offset , in the  form: 2006-10-25-10.52.43 */
-/*  offset is a string of the same form  */
-/*                                               0123456789012345678 */
-void
-getOffsetTimeStr( char *timeStr, const char *offSet ) {
-    time_t myTime;
-
-    myTime = time( NULL );
-    myTime += atoi( offSet );
 
-    snprintf( timeStr, TIME_LEN, "%d", ( uint ) myTime );
+/* Get the current time + offset , and store to timeStr (understood max length TIME_LEN) in epoch seconds.
+    OffSet is a string of the form nnnn<Unit> with Unit in {'','s','m','h','d','y'}.
+    An empty Unit field implies seconds. */
+int
+getOffsetTimeStr( char *timeStr, const char *offSet )
+{
+    long seconds_multiplier = 1L;
+    static const std::map<char,long> multiplier {
+        {'\0',1}, {'s',1}, {'m',60}, {'h',3600}, {'d',24*3600}, {'y',24*3600*365}
+    };
+    time_t myTime = time( NULL );
+    const char *lTrimOffSet = offSet;
+    while(isspace(*lTrimOffSet)) {++lTrimOffSet;}               // left-trim any whitespace
```

Comment:
Would it be worth checking for `nullptr` on the two inputs, or is that overkill in this situation?

Reply:
i had nullptr before but the man page says use NULL.  I know they're the same, but for C style code I generally write in C idioms ; )

Reply:
What I meant was that we dereference `lTrimOffset` (which is equivalent to `offSet` at first) without making sure it's a valid pointer and I didn't know if we wanted to check our inputs.

Reply:
ah good point